lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -164,7 +164,18 @@ class PullbackCloner::Implementation final
     assert(pbTupleTyple->getNumElements() == values.size() &&
            "The number of pullback tuple fields must equal the number of "
            "pullback struct element values");
-    auto res = pullbackTupleElements.insert({origBB, values});
+    auto res = pullbackTupleElements.insert({origBB, { values.begin(), values.end() }});
+    (void)res;
+    assert(res.second && "A pullback struct element already exists!");
+  }
+
+  void initializePullbackTupleElements(SILBasicBlock *origBB,
```

Comment:
This method seems to almost entirely duplicate the `SILInstructionResultArray` version of `initializePullbackTupleElements` above.

Would it make sense to rewrite one to call the other? (Perhaps not)

Reply:
Well, we need two versions here:
 - With `ArrayRef<SILArgument*>` for arguments
 - With `SILInstructionResultArray` (essentially `ArrayRef<SILValue>`) for tuple results

So, the extra conversions here and there very hairy, so I decided to keep the copy, as it is essentially a one-liner :)